server/schemas/schemaRoutes.js

```diff
@@ -4,6 +4,7 @@ module.exports = function(router) {
   router.get("/", schemaController.getSchemaNames);
   router.get("/:schemaName", schemaController.getEntriesMetadataForSchema);
   router.get("/:schemaName/:userID", schemaController.getData);
+  router.get("/template/:schemaName", schemaController.fetchTemplate);
```

Comment:
This will get clobbered by a schema named "template". Better to use `/:schemaName/template`, place the route before `/:schemaName/:userID`, and either ensure there's no possibility of assigning a `userID` of "template", or, better yet, ensure the routes never conflict.